pkg/controller/rpaasinstance/rpaasinstance_controller.go

```diff
@@ -404,6 +470,125 @@ func (r *ReconcileRpaasInstance) reconcileNginx(nginx *nginxv1alpha1.Nginx) erro
 	return err
 }
 
+func (r *ReconcileRpaasInstance) reconcileCacheHeaterCronJob(instance *v1alpha1.RpaasInstance, plan *v1alpha1.RpaasPlan) error {
+	foundCronJob := &batchv1beta1.CronJob{}
+	cronName := instance.Name + cacheHeaterCronJobSuffix
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: cronName, Namespace: instance.Namespace}, foundCronJob)
+	if err != nil && !k8sErrors.IsNotFound(err) {
+		return err
+	}
+
+	newestCronJob := newCronJob(instance, plan)
+	if k8sErrors.IsNotFound(err) {
+		return r.client.Create(context.TODO(), newestCronJob)
+	}
+
+	if !reflect.DeepEqual(foundCronJob.Spec, newestCronJob.Spec) {
+		return r.client.Update(context.TODO(), newestCronJob)
```

Comment:
We must set `newestCronJob.ResourceVersion = foundCronJob.ResourceVersion` before calling Update, otherwise it will fail.